### PR TITLE
initialize cpuInfo mbean lazily to reduce startup time

### PR DESCRIPTION
**Pull request by gjdeval**

### CODE REVIEW

File: dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java [hunk 1]
```diff
@@ -62,16 +62,6 @@ private CpuInfo() {
...
-        // initialize mbean required to get CPU usage info
-        osmx = ManagementFactory.getOperatingSystemMXBean();
-        mBeanServer = ManagementFactory.getPlatformMBeanServer();
-
-        try {
-            operatingSystemMbean = new ObjectName("java.lang", "type", "OperatingSystem");
-        } catch (MalformedObjectNameException e) {
-            FFDCFilter.processException(e, getClass().getName(), "CpuInfo<init>");
-        }
-
```
Comment (jhanders34) [inline dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java]: I suggest that you do something like this:

```
OperatingSystemMXBean osmx = ManagementFactory.getOperatingSystemMXBean();
operatingSystemMbean = osmx == null ? null : osmx.getObjectName();
```
Get rid of osmx as a class field and just have operatingSystemMbean be final.

File: dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java [hunk 2]
```diff
@@ -92,10 +82,28 @@ private CpuInfo() {
...
+                FFDCFilter.processException(e, getClass().getName(), "CpuInfo<init>");
+            }
+        }
+    }
+
     private double getSystemCPU() {
         final String OS_ATTRIBUTE_SYSTEM_CPU_LOAD = "SystemCpuLoad";
         double cpuUsage = -1;
 
+        if (osmx == null)
```
Comment (jhanders34) [inline dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java]: changes this code to look like:
```
if (operatingSystemMbean != null) {
    if (mbeanServer == null) {
        mbeanServer = ManagementFactory.getPlatformMBeanServer();
    }
    cpuUsage = (Double) mBeanServer.getAttribute(operatingSystemMbean, OS_ATTRIBUTE_SYSTEM_CPU_LOAD);
}
```
All that code goes into the try below.
Make the same change in the other method below and put it in the try.  

Also make mbeanServer a volatile field.

File: dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java [hunk 3]
```diff
@@ -39,13 +38,12 @@
...
 
-    private static CpuInfo instance;
+    private final static CpuInfo INSTANCE = new CpuInfo();
     private final int AVAILABLE_PROCESSORS;
 
     // For CPU usage calculation
-    private final OperatingSystemMXBean osmx;
-    private final MBeanServer mBeanServer;
-    private ObjectName operatingSystemMbean;
+    volatile MBeanServer mBeanServer = null;
```
Comment (jhanders34) [inline dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java]: Probably should be private.

File: dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java (hunk 1 above)
Comment (gjdeval) [reply dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java]: Done

File: dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java (hunk 2 above)
Comment (gjdeval) [reply dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java]: Nice code ... done

File: dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java (hunk 3 above)
Comment (gjdeval) [reply dev/com.ibm.ws.kernel.service/src/com/ibm/ws/kernel/service/util/CpuInfo.java]: Agreed, done

**Comment by gjdeval** [comment]
#build

**Pull request closed without merging**